docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -1167,16 +1170,403 @@ payload:
+including both sign-in and sign-up functionalities.
+
+PKCE flow for enhanced security
+-------------------------------
+
+The PKCE (Proof Key for Code Exchange) flow enhances security in server-to-server
+authentication by generating a unique verifier and its corresponding challenge.
+First, your server creates a 32-byte Base64 URL-encoded verifier, stores it in an
+HttpOnly cookie, hashes it with SHA256, and then encodes it to form the challenge.
+
+This implementation ensures enhanced security by preventing token leakage and is
+tailored specifically for server-to-server interactions.
+
+Add the following code to your FastAPI application to generate the PKCE:
+
+.. code-block:: python
+    :caption: app/auth.py
+
+    import secrets
+    import hashlib
+    import base64
+
+    def generate_pkce():
+        verifier = secrets.token_urlsafe(32)
+        challenge = hashlib.sha256(verifier.encode()).digest()
+        challenge_base64 = base64.urlsafe_b64encode(challenge).decode('utf-8').rstrip('=')
+        return verifier, challenge_base64
+
+User registration and authentication
+------------------------------------
+
+Next, we're going to create endpoints in FastAPI to handle user registration
+(sign-up) and user login (sign-in):
+
+**Sign-up endpoint**
+
+.. code-block:: python
+    :caption: app/auth.py
+
+    from fastapi import APIRouter, HTTPException, Request
+    from fastapi.responses import JSONResponse
+    import httpx
+
+    router = APIRouter()
+    
+    # Value should be:
+    # {protocol}://${host}:${port}/branch/${branch}/ext/auth/
+    EDGEDB_AUTH_BASE_URL = os.getenv('EDGEDB_AUTH_BASE_URL')
+
+    @router.post("/auth/signup")
+    async def handle_signup(request: Request):
+        body = await request.json()
+        email = body.get("email")
+        password = body.get("password")
+        name = body.get("name")
+
+        if not email or not password or not name:
+            raise HTTPException(status_code=400, detail="Missing email, password, or name.")
+
+        verifier, challenge = generate_pkce()
+        register_url = f"{EDGEDB_AUTH_BASE_URL}/register"
+        response = httpx.post(register_url, json={
+            "challenge": challenge,
+            "email": email,
+            "password": password,
+            "provider": provider,
+            "verify_url": "http://localhost:8000/auth/verify",
+        })
+
+        if response.status_code != 200 and response.status_code != 201:
+            return JSONResponse(status_code=400, content={"message": "Registration failed"})
+
+        response = JSONResponse(content={"message": "User registered"})
+        response.set_cookie(key="edgedb-pkce-verifier", value=verifier, httponly=True, secure=True, samesite='strict')
+        return response
+
+The sign-up endpoint sends a POST request to the EdgeDB Auth server to register
+a new user. It also generates a PKCE verifier and sets it as an HttpOnly cookie
+in the response.
```

Comment:
It doesn't need to set the verifier cookie in this endpoint because the whole PKCE session happens within this one endpoint, but we _do_ set the auth token cookie here.